### PR TITLE
ENH - Replace ``sklearn`` by ``celer``

### LINKED ISSUE #135: ENH use celer enet solver
Now that @Badr-Moufad implemented enet in celer and celer 0.7 is released, we can switch from sklearn solver to celer solver for sklearn. This should make the code faster !

### PR DESCRIPTION
**Pull request by Badr-MOUFAD**
This aims to use ``celer`` solver instead of ``sklearn``.

- [x] documentation (building plots)
- [x] unit test

closes #135

### CODE REVIEW

File: examples/plot_use_callback.py [hunk 1]
```diff
@@ -51,8 +51,8 @@
 alpha_max = np.max(np.abs(X[idx_train, :].T @ y[idx_train])) / len(idx_train)
 alpha0 = alpha_max / 10
 
-estimator = linear_model.Lasso(
-    fit_intercept=False, max_iter=1e5, warm_start=True)
+estimator = celer.Lasso(
+    fit_intercept=False, max_iter=100_000, warm_start=True)
```
Comment (mathurinm) [inline examples/plot_use_callback.py:55]: for celer the numbe rof iterations is not the same, you can put 50 here

File: expes/expe_enet/plot_enet_pred.py [hunk 2]
```diff
@@ -49,7 +49,7 @@
 
 tol = 1e-3
 
-estimator = linear_model.ElasticNet(
+estimator = celer.ElasticNet(
     fit_intercept=False, max_iter=1000, warm_start=True, tol=tol)
```
Comment (mathurinm) [inline expes/expe_enet/plot_enet_pred.py:53]: 1000 iter is a lot (it corresponds ton working set definitions)

File: expes/expe_fig_cross_val/main_crossval_elastic.py [hunk 3]
```diff
@@ -45,7 +45,7 @@
 max_evals = 25
 print("Starting path computation...")
 for algorithm in algorithms:
-    estimator = sklearn.linear_model.ElasticNet(
+    estimator = celer.ElasticNet(
         fit_intercept=False, max_iter=3_000, warm_start=True, tol=tol)
```
Comment (mathurinm) [inline expes/expe_fig_cross_val/main_crossval_elastic.py:49]: lots of iter

File: sparse_ho/optimizers/__init__.py [hunk 4]
```diff
@@ -2,6 +2,8 @@
 from sparse_ho.optimizers.adam import Adam
 from sparse_ho.optimizers.line_search import LineSearch
 
+from sparse_ho.ho import (hyperopt_wrapper, grad_search)
```
Comment (mathurinm) [inline sparse_ho/optimizers/__init__.py:5]: is this change needed ? not sure

File: sparse_ho/tests/test_grid_search.py [hunk 5]
```diff
@@ -32,7 +32,7 @@
 alphas = alpha_max * np.geomspace(1, 0.1)
 alpha_min = 0.0001 * alpha_max
 
-estimator = linear_model.Lasso(
+estimator = celer.Lasso(
     fit_intercept=False, max_iter=10000, warm_start=True)
```
Comment (mathurinm) [inline sparse_ho/tests/test_grid_search.py:36]: check iter everywhere

File: sparse_ho/tests/test_utils.py [hunk 6]
```diff
@@ -32,7 +32,7 @@
 max_iter = 1e5
 
 
-estimator = linear_model.Lasso(
+estimator = celer.Lasso(
     fit_intercept=False, max_iter=max_iter, warm_start=True)
```
Comment (mathurinm) [inline sparse_ho/tests/test_utils.py:36]: iter too large

**Comment by mathurinm** [comment]
Thanks Badr

@Klopfe @QB3 was there a particular reason for which we used sklearn's Lasso instead of celer's ?

**Comment by Klopfe** [comment]
> Thanks Badr
> 
> @Klopfe @QB3 was there a particular reason for which we used sklearn's Lasso instead of celer's ?

hmm... Not that I know of...

**Comment by mathurinm** [comment]
small failure here : https://github.com/QB3/sparse-ho/runs/6920007820?check_suite_focus=true#step:5:184 celer Logisticregression does not support multiclass so you need to keep sklearn's

I would use sklearn LogisticRegression everywhere in fact because it is quite fast (relying on liblinear etc)

**Review by mathurinm — APPROVED** [review]

**Review by agramfort — APPROVED** [review]
doc CI failure is unrelated and it's a QT problem I don't have much time to investigate.
